dex/pair/src/safe_price.rs

```diff
@@ -91,6 +91,72 @@ pub trait SafePriceModule:
         }
 
         let current_round = self.blockchain().get_block_round();
+        let round_save_interval = self.safe_price_round_save_interval().get();
+
+        // Handle the case where offset is 1 (immediate save)
+        if round_save_interval <= 1 {
+            self.handle_immediate_save(
+                current_round,
+                first_token_reserve,
+                second_token_reserve,
+                lp_supply,
+            );
+            return;
+        }
+
+        // Check if enough rounds have passed since last finalized observation for direct save
+        let safe_price_current_index = self.safe_price_current_index().get();
+        let last_observation = if safe_price_current_index > 0 {
+            let price_observations = self.price_observations();
+            price_observations.get(safe_price_current_index)
```

Comment:
Just use mapper directly, since you're not reusing it anyway.

```suggestion
            self.price_observations().get(safe_price_current_index)
```

Reply:
Updated.